### PR TITLE
[PKG-4138] Initial build 2.14.6

### PR DESCRIPTION
**Pull request by ViridianMelody**
pydantic-core 2.14.6

**Destination channel:** defaults

### Links

- [PKG-4138](https://anaconda.atlassian.net/browse/PKG-4138) 
- [Upstream repository](https://github.com/pydantic/pydantic-core)

### Explanation of changes:

- Forced `-headerpad_max_install_names` on osx-64 to fix a build failure. 
- Several test dependencies are not available on defaults, so tests are omitted.

[PKG-4138]: https://anaconda.atlassian.net/browse/PKG-4138?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/conda_build_config.yaml
Comment (JeanChristopheMorinPerso) [inline recipe/conda_build_config.yaml:1]: You can undo these changes

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,34 +1,33 @@
...
 
 build:
+  script_env:
+    - RUSTFLAGS=-C link-args=-headerpad_max_install_names  # [osx and x86_64]
+    - RUSTC_LOG=rustc_codegen_ssa::back::link=info  # [osx and x86_64]
   script:
-    # PyPy has weird sysconfigdata name
-    - rm -f $PREFIX/lib/pypy$PY_VER/_sysconfigdata.py  # [build_platform != target_platform and target_platform == "linux-ppc64le"]
-    - {{ PYTHON }} -m pip install . -vv
-    - cargo-bundle-licenses --format yaml --output THIRDPARTY.yml
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:16]: Keep this one and add `THIRDPARTY.yml` to the license_file field.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,34 +1,33 @@
...
 
 source:
   url: https://pypi.io/packages/source/p/pydantic-core/pydantic_core-{{ version }}.tar.gz
-  sha256: 1cac689f80a3abab2d3c0048b29eea5751114054f032a941a32de4c852c59cad
+  sha256: 1fd0c1d395372843fba13a51c28e3bb9d59bd7aebfeb17358ffaaa1e4dbbe948
 
 build:
+  script_env:
+    - RUSTFLAGS=-C link-args=-headerpad_max_install_names  # [osx and x86_64]
+    - RUSTC_LOG=rustc_codegen_ssa::back::link=info  # [osx and x86_64]
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,34 +1,33 @@
...
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/p/pydantic-core/pydantic_core-{{ version }}.tar.gz
-  sha256: 1cac689f80a3abab2d3c0048b29eea5751114054f032a941a32de4c852c59cad
+  sha256: 1fd0c1d395372843fba13a51c28e3bb9d59bd7aebfeb17358ffaaa1e4dbbe948
 
 build:
+  script_env:
+    - RUSTFLAGS=-C link-args=-headerpad_max_install_names  # [osx and x86_64]
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:15]:
```suggestion
    # on osx-64, we get an error from install_name_tool: changing install names or rpaths can't be redone for: <path> (for architecture x86_64) because larger updated load commands do not fit (the program must be relinked, and you may need to use -headerpad or -headerpad_max_install_names)
    # After some investigation, rustc doesn't seem to read our LDFLAGS
    - RUSTFLAGS=-C link-args=-headerpad_max_install_names  # [osx and x86_64]
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -45,9 +45,13 @@ test:
...
   home: https://github.com/pydantic/pydantic-core
   dev_url: https://github.com/pydantic/pydantic-core
+  doc_url: https://docs.pydantic.dev
   summary: Core validation logic for pydantic written in rust
+  description: |
+    This package provides the core functionality for pydantic validation and serialization.
   license: MIT
-  license_file: LICENSE
+  license_family: MIT
+  license_file: THIRDPARTY.yml
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
  license_file:
    - LICENSE
    - THIRDPARTY.yml
```

File: recipe/conda_build_config.yaml
Comment (JeanChristopheMorinPerso) [reply recipe/conda_build_config.yaml:1]: @ViridianMelody not sure why but this marked as outdated. You should undo your changes here.

**Review by SC426 — APPROVED** [review]

File: recipe/conda_build_config.yaml
Comment (JeanChristopheMorinPerso) [inline recipe/conda_build_config.yaml:1]: You can undo these changes.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,34 +1,36 @@
...
     - {{ compiler("c") }}
     - {{ compiler("rust") }}
-    - cargo-bundle-licenses
   host:
     - pip
     - python
     - maturin >=1,<2
     - typing-extensions  >=4.6.0,!=4.7.0
+    - wheel
+    - cargo-bundle-licenses
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: Move this in `build` (edit: build)

File: recipe/meta.yaml [hunk 6]
```diff
@@ -1,34 +1,36 @@
...
-    - maturin >=1,<2                         # [build_platform != target_platform]
     - {{ compiler("c") }}
     - {{ compiler("rust") }}
-    - cargo-bundle-licenses
   host:
     - pip
     - python
     - maturin >=1,<2
     - typing-extensions  >=4.6.0,!=4.7.0
+    - wheel
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
```
wheel is not required when `setuptools` is not used.

**Review by boldorider4 — APPROVED** [review]
LGTM, but please address linter issues (exact pinning and wheel missing).

File: recipe/meta.yaml [hunk 7]
```diff
@@ -1,34 +1,35 @@
...
     - {{ compiler("c") }}
     - {{ compiler("rust") }}
     - cargo-bundle-licenses
   host:
     - pip
     - python
-    - maturin >=1,<2
-    - typing-extensions  >=4.6.0,!=4.7.0
+    - maturin 1.3.1
+    - typing-extensions  4.9.0
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
    - maturin >=1,<2
    - typing-extensions  >=4.6.0,!=4.7.0
```
Adding hard/strict pin will just make our life harder in the future. You can keep the upstream pins.

**Review by JeanChristopheMorinPerso — CHANGES_REQUESTED** [review]

**Review by JeanChristopheMorinPerso — APPROVED** [review]
